### ISSUE TITLE
Exception gets lost

### ISSUE THREAD

**Issue by abbradar** [crowded/ply#10]
I didn't research what the root cause is, but in this example:

```
let testTask () =
    task {
        try
            do! Task.Yield ()
            failwith "Catch me"
        with
        | e ->
            failwith "Fail now"
        failwith "Test failed"
    }
```

Current output is "Test failed". Ply 0.2.2.

**Comment by NinoFloris** [comment]
Confirmed, this was an issue present in the new Combine implementation, fix is in 0.3.0 and will be verified shortly!

Regression tests for this and other cases have been added as well.

**Comment by abbradar** [comment]
Nice, thanks for the quick fix!
